tests/network/vmi_networking.go

```diff
@@ -677,20 +682,23 @@ var _ = SIGDescribe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:c
 				By("starting a tcp server")
 				tests.StartTCPServer(serverVMI, tcpPort)
 
+				gwIPAddress := customGWIPAddress
+
 				if networkCIDR == "" {
 					networkCIDR = api.DefaultVMCIDR
+					gwIPAddress = gatewayIPFromCIDR(networkCIDR)
 				}
 
 				By("Checking ping (IPv4) to gateway")
-				ipAddr := gatewayIPFromCIDR(networkCIDR)
```

Comment:
Now the custom networkCIDR is ignored.

Reply:
Because the current implementation of the tests doesn't support getting the GW address from a CIDR with leading zeros on go 1.17 - It fails at gatewayIPFromCIDR().
It could be solved by using ParseCIDRSloppy(), but as @EdDev had said, we don't want to legitimize usage of this function.

Reply:
ParseIPSloppy() and ParseCIDRSloppy() are used only once in the code (sanitizers.go)

Reply:
I find avoiding the `ParseCIDR` with this workaround is very confusing and hiding the real reason.
We don't really need an extra parameter for a custom gateway IP address. It is just a trick to avoid the `ParseCIDR`.
I prefer to be straight forward and use `ParseCIDRSloppy()`.

Reply:
Done.